neuron_morphology/pipeline/post_data_to_s3.py

```diff
@@ -0,0 +1,118 @@
+import os
+import boto3
+import json
+import configparser
+from zipfile import ZipFile, ZIP_DEFLATED
+from io import BytesIO
+from argschema import ArgSchemaParser
+from neuron_morphology.pipeline._schemas import InputParameters
+
+def zip_files(file_dict):
+    """
+    zip files into an archive in memory
+
+    Parameters
+    ---------------
+    file_dict: file name: file paths or file in bytes to be archived
+
+    Return
+    ---------------
+    archive: BytesIO obj
+
+    """
+
+    archive = BytesIO()
+
+    with ZipFile(archive, "a", ZIP_DEFLATED, False) as zip_file:
+        for name, file in file_dict.items():
+            if isinstance(file, str):
+                with open(file, 'rb') as fh:
+                    data = BytesIO(fh.read())
+                zip_file.writestr(name, data.getvalue())
+            elif isinstance(file, BytesIO):
+                zip_file.writestr(name, file.getvalue())
+            else:
+                raise TypeError("Invalid input file!")
+
+    return archive
+
+
+def post_object_to_s3(archive_data, archive_name, bucket, region, access_key_id=None, secret_access_key=None):
+    """
+    This zip files to an archive in memory and post it to S3 bucket
+
+    Parameters
+    ------------------
+    archive_data: the archive data
+    archive_name: the archive's name in s3
+    region: where the s3 bucket located
+    bucket: s3 bucket name or arn
+    access_key_id, secret_access_key: aws user's credentials
+    
+    Return
+    ------------------
+    True if successful
+    
+    """
+    if access_key_id and secret_access_key:
+        s3_client = boto3.client(
+            's3',
+            aws_access_key_id=access_key_id,
+            aws_secret_access_key=secret_access_key,
+            region_name=region
+        )
+    else:
+        s3_client = boto3.client('s3', region_name=region)
+
+    response = s3_client.put_object(Bucket=bucket, 
```

Comment:
This is confusing to me: 
This looks as if access_key_id and secret_access_key are provided then we use them, but if they are not provided then we do not use them. But my understanding is that this is not what is actually happening. Boto is still looking for these parameters but elsewhere. It would be better to explicitly provide `access_key_id` and `secret_access_key` to this function and avoid the implicit specification of credentials.

Reply:
https://boto3.amazonaws.com/v1/documentation/api/latest/guide/configuration.html

If credentials_file are not provided explicitly, boto3 will find them in ~/.aws/credentials (/allen/aibs/technology/production/.aws/credentials) or environment setting with AWS_SHARED_CREDENTIALS_FILE to your credentials file.

Reply:
how would boto know to look at this path /allen/aibs/technology/production/.aws/credentials?